### PR TITLE
Implement a separate cli for wicked to NM migration

### PR DESCRIPTION
**Pull request by jcronenberg**
## Problem

The migration should be independent of agama and thus having it as a subcommand in the agama binary doesn't make sense.

## Solution

Write a separate cli that handles everything the migration needs.
**Note:** When running the migration in a container it will produce errors as it tries to activate the connection which it can't because the interfaces aren't available (This error happened also previously but because agama-cli doesn't implement the logger it just didn't show up). In the future it may make sense to implement a way for adding a connection in agama without activating it.

## Testing

- *Tested manually*

### CODE REVIEW

File: rust/agama-migrate-wicked/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Migrate wicked
+This project creates a `migrate-wicked` binary which is able to parse wicked xml configs and send them to a NetworkManager dbus service.
+## Architecture
+`migrate-wicked` uses agama as a library to communicate the parsed network state to NetworkManager
+but the binary is completely independent of any agama services and can be run standalone
+## Obtaining wicked xml config
+A wicked config xml can be generated by running `wicked show-config > wicked.xml` which can then be passed to `migrate-wicked`
+## Testing
+Running the migration on a live system isn't currently recommended. To test it's recommended to use a container
+(instructions how to set one up can be found [here](https://github.com/openSUSE/agama/blob/master/rust/agama-cli/doc/backend-for-testing.md))
```
Comment (cfconrad) [inline rust/agama-migrate-wicked/README.md:10]: Just a comment: I think this container is actually quite heavy for our needs. As we simply need a container with NetworkManager installed, or?

File: rust/agama-migrate-wicked/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                Format::Json => serde_json::to_string(&interfaces)?,
+                Format::PrettyJson => serde_json::to_string_pretty(&interfaces)?,
+                Format::Yaml => serde_yaml::to_string(&interfaces)?,
+                Format::Text => format!("{:?}", interfaces),
+            };
+            println!("{}", output);
+            Ok(())
+        }
+        Commands::Migrate { path } => {
+            migrate(path).await.unwrap();
```
Comment (cfconrad) [inline rust/agama-migrate-wicked/src/main.rs:69]: maybe?
```suggestion
            migrate(path).await?;
```

File: rust/agama-migrate-wicked/README.md (hunk 1 above)
Comment (jcronenberg) [reply rust/agama-migrate-wicked/README.md:10]: You're right. I think for now and just for testing, it's not that important but I guess we could write a simpler script that skips all the unnecessary steps. Though for now I'd keep the image the same and maybe later build our own minimal container. WDYT?

File: rust/agama-migrate-wicked/src/main.rs (hunk 2 above)
Comment (jcronenberg) [reply rust/agama-migrate-wicked/src/main.rs:69]:
```
`dyn StdError` cannot be sent between threads safely
```

**Review by cfconrad — APPROVED** [review]
LGTM

**Pull request merged**